aptos-move/framework/aptos-experimental/sources/trading/market/market.move

```diff
@@ -717,7 +716,9 @@ module aptos_experimental::market {
                     }
                 );
                 // If the maker is invalid cancel the maker order and continue to the next maker order
-                self.order_book.cancel_order(maker_address, maker_order_id);
+                if (maker_order.get_remaining_size() != 0) {
```

Comment:
When can a maker be canceled with a non zero remaining size?

Reply:
Not sure I understand, but we want to cancel the maker order, part of the maker order which is removed from the orderbook is automatically cancelled for there is a part that might be remaining on the orderbook and we need to check if the remaining size on the order book is non zero before cancelling it. This change is needed because I changed the `cancel_order` API to fail if the order doesn't exists.